src/Microsoft.ComponentDetection.Contracts/TypedComponent/VcpkgComponent.cs

```diff
@@ -88,15 +88,17 @@ private void SetGitRepoProperties()
          * "git+https://github.com/leethomason/tinyxml2@9.0.0"
          * "git+https://github.com/Microsoft/vcpkg#ports/nlohmann-json"
         */
-        var locationArr = this.DownloadLocation.Split('/');
-        if (!string.IsNullOrEmpty(locationArr[2]))
-        {
-            this.GitRepositoryOwner = locationArr[2];
-        }
 
-        if (!string.IsNullOrEmpty(locationArr[3]))
+        if (Uri.TryCreate(this.DownloadLocation, UriKind.Absolute, out var tempUri))
         {
-            this.GitRepositoryName = locationArr[3].TakeWhile(ch => char.IsLetterOrDigit(ch)).ToString();
+            var locationArr = tempUri.Segments;
+            if (locationArr.Length < 3)
+            {
+                return;
+            }
+
+            this.GitRepositoryOwner = locationArr[1].TrimEnd('/');
```

Comment:
These properties are not first class for Vcpkg, downstream clients should capture this themselves if desirable. This data is encoded in native DownloadLocation.